File: codes/mathematica/p001.m
(* 
 * Solution to Project Euler problem 1
 * Copyright (c) Sankha Saha. All rights reserved.
 * 
 *  
 *   
 *)


(* Computers are fast, so we can implement this solution directly without any clever math. *)
Total[Select[Range[999], Function[x, Mod[x, 3] == 0 || Mod[x, 5] == 0]]]

File: codes/Haskell/p001.hs
{- 
 - Solution to Project Euler problem 1
 - Copyright (c) Sankha Saha. All rights reserved.
 - 
 -  
 -   
 -}


-- Computers are fast, so we can implement this solution directly without any clever math.
main = putStrLn (show ans)
ans = sum [x | x <- [0..999], (mod x 3) == 0 || (mod x 5) == 0]
